miden-lib/tests/test_account.rs

```diff
@@ -306,3 +306,93 @@ fn test_set_item() {
     )
     .unwrap();
 }
+
+#[test]
+fn test_is_faucet_procedure() {
+    let test_cases = vec![
+        ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN,
+        ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_OFF_CHAIN,
+        ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN,
+        ACCOUNT_ID_NON_FUNGIBLE_FAUCET_OFF_CHAIN,
+    ];
+
+    for account_id in test_cases.iter() {
+        let account_id = AccountId::try_from(*account_id).unwrap();
+
+        // assembly codes that checks if an account is a fauct
+        let code = format!(
+            "
+        use.miden::sat::account
+
+        begin
+            # push the account id on to the stack
+            push.{account_id}
+
+            # execute is_faucet procedure
+            exec.account::is_faucet
+
+            # assert it matches expected result
+            eq.{expected} assert
+        end
+    ",
+            account_id = *account_id,
+            expected = if account_id.is_faucet() { 1 } else { 0 },
+        );
+
+        let _process = run_within_tx_kernel(
+            "",
+            &code,
+            StackInputs::default(),
+            MemAdviceProvider::default(),
+            None,
+            None,
+        )
+        .unwrap();
+    }
+}
+
+#[test]
+fn test_authenticate_procedure() {
+    let inputs = mock_inputs();
+
+    let test_cases = vec![
+        (inputs.account().code().method_tree().get_leaf(0).unwrap(), true),
+        (inputs.account().code().method_tree().get_leaf(1).unwrap(), true),
+        (Word::default(), false),
```

Comment:
This probably isn't a great negative test case because we could easily authenticate `Word::default()` using one of the empty leafs in the account method tree. That begs a different question.  We need to ensure that it is practically infeasible to produce a malicious method that hashes to `Word::default()`.

Reply:
Do you think it is warranted to add these checks?  Access checks are likely to be relatively commonly invoked so there may be some overhead here.  If the system relies on the assumption of a secure hash function then it probably doesn't make sense to add the extra check.  What do you think?

Reply:
I don't think they are necessary.